### PR TITLE
Save each user's associated wallet, use them to count for ticket purchase quota

### PR DESCRIPTION
**Pull request by pointmekin**
## Description

- update user DTO, schema, and interface to add associated wallets
- update associate wallets each time a wallet is used to buy a ticket
- check quota from all wallets associated to a user's account

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/user/dto/user.dto.ts [hunk 1]
```diff
@@ -131,3 +135,14 @@ export class UpdateVerificationStatusDTO {
   @IsString()
   verificationStatus: Verification;
 }
+
+@Expose()
+export class updateAssociatedWalletAddressDTO {
```
Comment (chuwong35122) [inline src/user/dto/user.dto.ts:140]: Make it Uppercase

File: src/user/dto/user.dto.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ export class CreateCustomerUserDTO {
 
   @ApiProperty()
   profileImage: File | string;
+
+  @ApiProperty({ type: [String] })
```
Comment (chuwong35122) [inline src/user/dto/user.dto.ts:42]: use `0x${string}`[] ?

File: src/ticket/ticket.service.ts [hunk 3]
```diff
@@ -470,9 +546,16 @@ export class TicketService {
     address: string,
     ticketCollectionId: string,
     ticketType: string,
+    userId: string,
     smartContractTicketId?: number,
   ): Promise<TicketQuotaCheckResultDTO> {
-    const ownedTicketsResponse: MyTicketsDTO = await this.getTicketsOfUser(address);
+    // Get the associated wallets of the user
+    const user: CustomerUser = (await this.userService.findById(userId)) as CustomerUser;
```
Comment (chuwong35122) [inline src/ticket/ticket.service.ts:553]: typing too redundant

File: src/ticket/ticket.service.ts [hunk 4]
```diff
@@ -203,6 +204,63 @@ export class TicketService {
     }
   }
 
+  // Get all tickets belonging to a person including those they own and those they are selling
+  async getTicketsOfUserFromMultipleWalletAddresses(walletAddressList: string[]): Promise<MyTicketsDTO> {
+    try {
+      const ticketCollections = await this.ticketCollectionModel.find().exec();
```
Comment (chuwong35122) [inline src/ticket/ticket.service.ts:210]: This will find every document, right? Is there a better operation?

**Review by chuwong35122 — CHANGES_REQUESTED** [review]

File: src/ticket/ticket.service.ts (hunk 4 above)
Comment (pointmekin) [reply src/ticket/ticket.service.ts:210]: Good point. This one can take ticket collection id and just search from that ticket collection only. Thanks for pointing out.

**Pull request merged**